.dotnet/src/Custom/Chat/ChatClient.cs

```diff
@@ -1,17 +1,21 @@
+using OpenAI.Internal.Models;
 using System;
 using System.ClientModel;
 using System.ClientModel.Primitives;
 using System.Collections.Generic;
-using System.Text;
+using System.Net;
 using System.Threading.Tasks;
 
 namespace OpenAI.Chat;
 
-/// <summary> The service client for the OpenAI Chat Completions endpoint. </summary>
 public partial class ChatClient
 {
-    private readonly OpenAIClientConnector _clientConnector;
-    private Internal.Chat Shim => _clientConnector.InternalClient.GetChatClient();
+    protected readonly string _model;
+    protected readonly ClientPipeline _pipeline;
+    protected readonly Uri _endpoint;
+
+    /// <summary> The HTTP pipeline for sending and receiving REST requests and responses. </summary>
+    public virtual ClientPipeline Pipeline => _pipeline;
```

Comment:
does this need to be virtual? I thought only the method that creates the pipeline will be virtual.

Reply:
You're fully correct. Removing virtual; we really wouldn't want this to be overriden.